pkg/controllers/korrel8r/components.go

```diff
@@ -0,0 +1,122 @@
+)
+
+func korrel8rComponentReconcilers(korrel8rDeploy *appsv1.Deployment, korrel8rSvc *corev1.Service, korrel8rCfg Korrel8rConfiguration, namespace string) []reconciler.Reconciler {
+
+	return []reconciler.Reconciler{
+		reconciler.NewUpdater(newKorrel8rService(korrel8rObjName, namespace), korrel8rSvc),
+		reconciler.NewUpdater(newKorrel8rDeployment(korrel8rObjName, namespace, korrel8rCfg), korrel8rDeploy),
+	}
+}
+
+func newKorrel8rDeployment(name string, namespace string, korrel8rCfg Korrel8rConfiguration) *appsv1.Deployment {
+	deploy := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    componentLabels(name),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: componentLabels(name),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels:    componentLabels(name),
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: name + serviceAccountSuffix,
+					Containers: []corev1.Container{
+						{
+							Name:    name,
+							Image:   korrel8rCfg.Image,
+							Command: []string{"korrel8r", "web", "--config", "/etc/korrel8r/openshift-svc.yaml"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8443,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								RunAsNonRoot:             ptr.To(true),
+								AllowPrivilegeEscalation: ptr.To(false),
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{
+										"ALL",
+									},
+								},
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: corev1.SeccompProfileTypeRuntimeDefault,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return deploy
+}
+
+func newKorrel8rService(name string, namespace string) *corev1.Service {
+	annotations := map[string]string{
+		"service.alpha.openshift.io/serving-cert-secret-name": name,
+	}
+
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Labels:      componentLabels(name),
+			Annotations: annotations,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port:       port,
+					Name:       "web",
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt32(port),
+				},
+			},
+			Selector: componentLabels(name),
+			Type:     corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func componentLabels(name string) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/instance":   name,
+		"app.kubernetes.io/part-of":    "Korrel8r",
```

Comment:
Need a consistent interpretation for these app.k8s.io labels.
E.g. is korrel8r part-of COO or part-of itself? Is korrel8r a component or does it have components?
Is instance a unique name within a namespace or unique cluster-wide?

The k8s docs are a bit ambiguous. I'm not so concerned about finding the "right" answer, more about being consistent with the rest of COO.